pkg/services/query/query.go

```diff
@@ -149,6 +151,19 @@ func (s *Service) handleQueryData(ctx context.Context, user *models.SignedInUser
 		req.Headers[k] = v
 	}
 
+	if parsedReq.httpRequest != nil && parsedReq.httpRequest.Header.Get("Cookie") != "" && ds.JsonData != nil {
+		keepCookieNames := []string{}
+
+		if keepCookies := ds.JsonData.Get("keepCookies"); keepCookies != nil {
```

Comment:
This seems like a kinda big feature to stick on the untyped jsonData

Where is this used?  Is it something that is really set explicitly from the UI?  It seems more like a setting attached to the whole plugin than something that would selectively apply to some instances

Reply:
It's an oldie part of the datasource HTTP settings for years: https://github.com/grafana/grafana/blob/8fa8c9dc8b6922110d154705b9d822bb5319819d/packages/grafana-ui/src/components/DataSourceSettings/DataSourceHttpSettings.tsx#L172-L186